src/Stack/Config/Nix.hs

```diff
@@ -32,6 +34,12 @@ nixOptsFromMonoid NixOptsMonoid{..} os = do
         nixShellOptions = fromFirst [] nixMonoidShellOptions
                           ++ prefixAll (T.pack "-I") (fromFirst [] nixMonoidPath)
         nixAddGCRoots   = fromFirst False nixMonoidAddGCRoots
+    nixEnable <-
+      if nixEnable0 && osIsWindows
```

Comment:
I can't say exactly what strikes me about this, but I find this code particularly elegant.

Reply:
Thanks! :)